src/ccpp_hash_table.F90

```diff
@@ -0,0 +1,520 @@
+!!XXgoldyXX: To do, statistics output
+module ccpp_hash_table
+
+   use ccpp_hashable, only: ccpp_hashable_t
+
+   implicit none
+   private
+
+   !
+   !  Constants used in hashing function gen_hash_key.
+   !
+
+   integer, parameter :: gen_hash_key_offset = 21467 ! z'000053db'
+
+   integer, parameter :: tbl_max_idx = 15
+   integer, parameter, dimension(0:tbl_max_idx) :: tbl_gen_hash_key =         &
+        (/ 61, 59, 53, 47, 43, 41, 37, 31, 29, 23, 17, 13, 11, 7, 3, 1 /)
```

Comment:
Why no 5 in the list? It's a prime number if I remember correctly. But I don't understand that `gen_hash_key` yet (need more time).

Reply:
Hmm, I guess I lost the comment that I swiped this algorithm from perl (with minor mods).